### PR TITLE
fix(ESF): Update items on 1st filter column

### LINKED ISSUE #10904: Excel Style Filter showing incorrect options
## Description  
I was working on my grid and applying some filters.
Suddenly a set of filters rendered my view blank. I had to reset my filters to load the grid data.
I investigated the excel style filter and some issues.

 * igniteui-angular version: 12.3.0
 * browser: Chrome

## Steps to reproduce  

1. Apply some filters on a column. For eg. Column A
2. Apply some filters on another column. For eg. Column B
3. Reopen the filter for Column A. It still has some options which should not be there. Applying those options renders the grid to show no data.

## Result  
What is the actual result after following the steps to reproduce?

https://user-images.githubusercontent.com/79090369/150990904-1b8b5a6b-8796-4807-abc3-6c50ac6c297d.mp4

## Expected result  
What is the expected result after following the steps to reproduce?
The filters should automatically remove invalid values.

## Attachments  
Attach a sample if available, and screenshots, if applicable.  
After applying the invalid filter the filter menu refreshes.
As shown in the image below the option for Blanks is not available now, but as shown in the video it was still there.
image

### PR DESCRIPTION
**Pull request by gedinakova**
Closes #10904   

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Review by onlyexeption — APPROVED** [review]

**Comment by kdinev** [comment]
@gedinakova Noticed the `do not merge` right after I clicked merge. Do you want me to revert the commit?

**Comment by gedinakova** [comment]
> @gedinakova Noticed the `do not merge` right after I clicked merge. Do you want me to revert the commit?

No, I will be looking more into this today and will make a new PR when ready, no worries.

**Pull request merged**
